### PR TITLE
added new headers pkg as new source of truth for common request headers

### PR DESCRIPTION
**Pull request by daiLlew**
### What

Added new `headers` package to replace duplicated & inconsistent header utilities in `go-ns`.
Added functions for `Get/Set` `Collection-Id` header plus unit tests.

### CODE REVIEW

File: headers/headers.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
+// headers package provides header name constants and get/set functions for commonly used http headers in the
+// dp-api-clients-go repo. Package replaces go-ns lib and should be treated as the single source of truth
```
Comment (dandwal) [inline headers/headers.go]: Likely worth us updating the readme in go-ns to mention this too

File: headers/headers.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+	headerValue := req.Header.Get(headerName)
+	if len(headerValue) == 0 {
+		return "", ErrHeaderNotFound
+	}
+
+	return headerValue, nil
+}
+
+func setRequestHeader(req *http.Request, headerName string, headerValue string) error {
```
Comment (dandwal) [inline headers/headers.go]: Just a suggestion not an issue.
Alternativly could have
```suggestion
func setRequestHeader(req *http.Request, headerName, headerValue string) error {
```

Comment (daiLlew) [reply headers/headers.go]: Personally not a fan of that style

File: headers/headers.go (hunk 1 above)
Comment (daiLlew) [reply headers/headers.go]: Its next on my list 😉

**Pull request merged**